crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,366 @@
+    JsBinaryOperator, JsLogicalExpression, JsParenthesizedExpression, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_lint_rule! {
+    /// Disallow unnecessary length checks within logical expressions.
+    ///
+    /// - `Array#some()` returns `false` for an empty array. There is no need to check if the array is not empty.
+    /// - `Array#every()` returns `true` for an empty array. There is no need to check if the array is empty.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if (array.length === 0 || array.every(Boolean));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (array.length !== 0 && array.some(Boolean));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (array.length > 0 && array.some(Boolean));
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const isAllTrulyOrEmpty = array.length === 0 || array.every(Boolean);
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (array.every(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// if (array.some(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// const isAllTrulyOrEmpty = array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// const isAllTrulyOrEmpty = array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// if (array.length === 0 || anotherCheck() || array.every(Boolean));
+    /// ```
+    ///
+    /// ```js
+    /// const isNonEmptyAllTrulyArray = array.length > 0 && array.every(Boolean);
+    /// ```
+    ///
+    /// ```js
+    /// const isEmptyArrayOrAllTruly = array.length === 0 || array.some(Boolean);
+    /// ```
+    ///
+    pub NoUselessLengthCheck {
+        version: "next",
+        name: "noUselessLengthCheck",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintUnicorn("no-useless-length-check")],
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum ErrorType {
+    UselessLengthCheckWithSome,
+    UselessLengthCheckWithEvery,
+}
+
+/// Whether the node is a descendant of a logical expression.
+fn is_logical_exp_descendant(node: &AnyJsExpression, operator: JsSyntaxKind) -> bool {
+    let Some(parent) = node.syntax().parent() else {
+        return false;
+    };
+    parent
+        .ancestors()
+        .find_map(|ancestor| {
+            if let Some(logical_exp) = JsLogicalExpression::cast(ancestor.clone()) {
+                return logical_exp
+                    .operator_token()
+                    .ok()
+                    .map(|token| token.kind() == operator)
+                    .or(Some(false));
+            }
+            (!JsParenthesizedExpression::can_cast(ancestor.kind())).then_some(false)
+        })
+        .unwrap_or(false)
+}
+
+/// Extract the expressions that perform length comparisons corresponding to the errors you want to check.
```

Comment:
I find it difficult to understand the documentation; maybe you should provide an example. Perhaps because you call the second argument "error", you should see a better name, like the one I suggested. 

Also, it would be nice to provide a code example in the docs.